src/org/rust/lang/core/resolve/scope/RustResolveScope.kt

```diff
@@ -1,13 +1,12 @@
 package org.rust.lang.core.resolve.scope
 
-import com.intellij.psi.PsiElement
 import org.rust.lang.core.psi.RustCompositeElement
+import org.rust.lang.core.psi.RustDeclaringElement
 import org.rust.lang.core.psi.RustNamedElement
-import org.rust.lang.core.psi.RustPatVar
 import org.rust.lang.core.resolve.RustResolveEngine
 
 public interface RustResolveScope : RustCompositeElement {
-    fun listDeclarations(before: PsiElement): List<RustPatVar>
+    fun getDeclarations(): Collection<RustDeclaringElement>
```

Comment:
Are you sure that `Collection` is enougth here? At least current implementation depends on the order, so a `List` may be more apropriet.